### PR TITLE
feature/challenge/arrays/exercises/28_29

### PR DESCRIPTION
**Pull request by EdgarAyllon**

### CODE REVIEW

File: course/frontend/arrays/scripts/main.js [hunk 1]
```diff
@@ -213,3 +213,76 @@ function flat(arr, newArray) {
...
+ * Function removeFalsy
+ * Returns one numeric array without empty,
+ * @param {array} first argument 
+ * @returns {array}  
+ */
+
+var unionArray, removeFalsy;
+
+(function(){
+'use strict'
```
Comment (manuelbarzi) [inline course/frontend/arrays/scripts/main.js]: ADD ";" at the end of this statement (we are in ES5 ,)

File: course/frontend/arrays/scripts/main.js [hunk 2]
```diff
@@ -213,3 +213,76 @@ function flat(arr, newArray) {
...
+
+*/
+
+/** 
+ * Function unionArray 
+ * Returns one numeric merged array without duplicated values.
+ * @param {array} first argument
+ * @param {array} second argument
+ * @returns {array} numeric values.
+ */
```
Comment (manuelbarzi) [inline course/frontend/arrays/scripts/main.js]: EACH function should be tied to its doc and implementation. so, following this rule i encourage you to split it into two, that is:

/**
 * DOC
 */
var whatever;
(function() {
// impl of whatever here
})();

/**
 * DOC 2
 */
var anotherThing;
(function() {
// impl of anotherThing here
})();

the reason why doing so is as good practice for documentation (puting the doc tied to the implementation) and splitting the implementation too, so later on you can easily move each one to its own file, independently ,)

SAME with tests (console.logs in this case)

File: course/frontend/arrays/scripts/main.js (hunk 1 above)
Comment (EdgarAyllon) [reply course/frontend/arrays/scripts/main.js]: yes, you are right. I did not notice about this. A

File: course/frontend/arrays/scripts/main.js [hunk 3]
```diff
@@ -213,3 +213,80 @@ function flat(arr, newArray) {
...
```
Comment (EdgarAyllon) [inline course/frontend/arrays/scripts/main.js]: Changes done.

File: course/frontend/arrays/scripts/main.js [hunk 4]
```diff
@@ -213,3 +213,80 @@ function flat(arr, newArray) {
...
+'use strict';
```
Comment (manuelbarzi) [inline course/frontend/arrays/scripts/main.js]: MOVE 'use strict'; back to each self-invoking function (each one may have it if you apply strict mode to your own code, something that i encourage you always to do). the point on this is that you make strict coding there inside, and do not force outer code to work strictly  (imagine third-party code around, not working in that way). so, in brief, this is totally right:

var func1;
(function() {
  'use strict';
 ... impl func1
})();

var func2;
(function() {
  'use strict';
 ... impl func2
})();

File: course/frontend/arrays/scripts/main.js [hunk 5]
```diff
@@ -213,3 +213,80 @@ function flat(arr, newArray) {
...
+/*
+### unionArray
+
+Write a JavaScript program to compute the union of two arrays.
+Sample Data :
+
+    console.log(unionArray([1, 2, 3], [100, 2, 1, 10]));
+    [1, 2, 3, 10, 100]
+
+### removeFalsy
```
Comment (manuelbarzi) [inline course/frontend/arrays/scripts/main.js]: MOVE this "requirement / explanation" about "removeFalsy" just above its doc / comment ,)

**Pull request merged**